src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/Harness/EventTestHarness.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System.Diagnostics.Tracing;
+#if USE_ETW
+using Microsoft.Diagnostics.Tracing.Session;
+#endif
+using Xunit;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+//Our tests shouldn't be parallelized. They often use the same EventSource, and only one instance can be active at once.
+[assembly: CollectionBehavior(DisableTestParallelization = true)]
+
+namespace BasicEventSourceTests
+{
+    /// <summary>
+    /// The EventTestHarness class knows how to run a set of single-event tests on either
+    /// the ETW pipeline or the EventListener pipeline.  
+    /// 
+    /// Basically you make up a bunch of SubTests and then hand it to this harness
+    /// to run them in bulk.   
+    /// </summary>
+    public static class EventTestHarness
+    {
+        /// <summary>
+        /// Runs a series of tests 'tests' using the listener (either an ETWListener or an EventListenerListener) on 
+        /// an EventSource 'source' passing it the filter parameters=options (by default source turn on completely
+        /// 
+        /// Note that this routine calls Dispose on the listener, so it can't be used after that.  
+        /// </summary>
+        public static void RunTests(List<SubTest> tests, Listener listener, EventSource source, FilteringOptions options = null)
+        {
+            Console.WriteLine("**** Starting Testing of " + tests.Count + " tests with Listener " + listener.GetType().Name);
```

Comment:
`Console.WriteLine` in tests is useless and [clutter CI output](http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/windows_nt_debug_prtest/50/console)